plugins/inputs/diskio/diskio_linux_test.go

```diff
@@ -19,19 +19,31 @@ S:foo/bar/devlink1
 `)
 
 // setupNullDisk sets up fake udev info as if /dev/null were a disk.
-func setupNullDisk(t *testing.T) func() error {
-	td, err := ioutil.TempDir("", ".telegraf.TestDiskInfo")
+func setupNullDisk(t *testing.T, s *DiskIO, devName string) func() error {
+	td, err := ioutil.TempFile("", ".telegraf.DiskInfoTest")
 	require.NoError(t, err)
 
-	origUdevPath := udevPath
+	if s.infoCache == nil {
+		s.infoCache = map[string]diskInfoCache{}
```

Comment:
I'd prefer
```suggestion
		s.infoCache = make(map[string]diskInfoCache, 0)
```
but it seems cosmetics...

Reply:
Doing this introduced an error:
```plugins/inputs/diskio/diskio_linux.go:30:46: map[string]diskInfoCache literal is not a type```

Also this line wasn't touched by me at all so it feels a bit out of place to be changing this.